ports/esp32/uart.c

```diff
@@ -27,21 +27,59 @@
  */
 
 #include <stdio.h>
-
+#include "uart.h"
 #include "driver/uart.h"
 #include "soc/uart_periph.h"
 
 #include "py/runtime.h"
 #include "py/mphal.h"
 
+#ifndef MICROPY_HW_UART_REPL_BAUD
+#define MICROPY_HW_UART_REPL_BAUD   115200
+#endif
+
 STATIC void uart_irq_handler(void *arg);
 
 void uart_init(void) {
+    uart_config_t uartcfg = {
+        .baud_rate = MICROPY_HW_UART_REPL_BAUD,
+        .data_bits = UART_DATA_8_BITS,
+        .parity = UART_PARITY_DISABLE,
+        .stop_bits = UART_STOP_BITS_1,
+        .flow_ctrl = UART_HW_FLOWCTRL_DISABLE,
+        .rx_flow_ctrl_thresh = 0
+    };
+    uart_param_config(UART_NUM_0, &uartcfg);
```

Comment:
Might be worth making `UART_NUM_0` a #define macro as well, eg `MICROPY_HW_UART_REPL` would match stm32.